From 9516c507736fac697589ecdacdecee28d42b0247 Mon Sep 17 00:00:00 2001
From: Olli Helenius <liff@iki.fi>
Date: Sun, 3 Sep 2023 12:33:53 +0300
Subject: [PATCH] Add support for tree-sitter modes

---
 2 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/smartparens-config.el b/smartparens-config.el
index 7bfcfe0..2540a4b 100644
--- a/smartparens-config.el
+++ b/smartparens-config.el
@@ -117,6 +117,8 @@ ID, ACTION, CONTEXT."
 ;; macro, you MUST supply the major mode argument.
 
 (eval-after-load 'cc-mode                  '(require 'smartparens-c))
+(eval-after-load 'c-ts-mode                '(require 'smartparens-c))
+(eval-after-load 'c++-ts-mode              '(require 'smartparens-c))
 (eval-after-load 'clojure-mode             '(require 'smartparens-clojure))
 (eval-after-load 'crystal-mode             '(require 'smartparens-crystal))
 (eval-after-load 'elixir-mode              '(require 'smartparens-elixir))
diff --git a/smartparens-python.el b/smartparens-python.el
index dbe6333..d94e459 100644
diff --git a/smartparens.el b/smartparens.el
index bf99f5f..a2f830b 100644
--- a/smartparens.el
+++ b/smartparens.el
@@ -606,6 +606,8 @@ Symbol is defined as a chunk of text recognized by
 (defcustom sp-c-modes '(
                         c-mode
                         c++-mode
+                        c-ts-mode
+                        c++-ts-mode
                         )
   "List of C-related modes."
   :type '(repeat symbol)
-- 
2.41.0

